File: sandbox-runner/patterns/continue-as-new/python/starter.py
import asyncio
import time

from temporalio.client import Client

from shared import TASK_QUEUE, WORKFLOW_ID_PREFIX
from workflows import DataProcessorWorkflow


async def main() -> None:
    client = await Client.connect("localhost:7233")
    workflow_id = f"{WORKFLOW_ID_PREFIX}-{int(time.time() * 1000)}"
    handle = await client.start_workflow(
        DataProcessorWorkflow.run,
        args=["", 0],
        id=workflow_id,
        task_queue=TASK_QUEUE,
    )
    print(f"Started workflow: {workflow_id}")

    total = await handle.result()
    print(f"Final result: processed {total} records")
    print(
        f"Open the Temporal UI and search for '{workflow_id}' to see the Continue-As-New chain."
    )


if __name__ == "__main__":
    asyncio.run(main())

File: sandbox-runner/patterns/continue-as-new/typescript/starter.ts
import { Client, Connection } from "@temporalio/client";

import { TASK_QUEUE, WORKFLOW_ID_PREFIX } from "./shared";
import { dataProcessorWorkflow } from "./workflows";

async function main(): Promise<void> {
  const connection = await Connection.connect();
  try {
    const client = new Client({ connection });
    const workflowId = `${WORKFLOW_ID_PREFIX}-${Date.now()}`;
    const handle = await client.workflow.start(dataProcessorWorkflow, {
      args: ["", 0],
      taskQueue: TASK_QUEUE,
      workflowId,
    });
    console.log(`Started workflow: ${workflowId}`);

    const total = await handle.result();
    console.log(`Final result: processed ${total} records`);
    console.log(`Open the Temporal UI and search for '${workflowId}' to see the Continue-As-New chain.`);
  } finally {
    await connection.close();
  }
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
